api/src/main/java/org/openmrs/module/fgh/mpi/MpiIntegrationProcessor.java

```diff
@@ -138,6 +138,12 @@ public Map<String, Object> process(Integer patientId, DatabaseEvent e) throws Ex
 				//And we will need to be aware of placeholder rows
 				Map<String, Object> generated = FhirUtils.buildPatient(id, patientVoided, personDetails, mpiPatient);
 				
+				if (generated.get("name") == null) {
+					log.warn("Skipping patient with no name");
```

Comment:
I want to disagree here, if a patient has no name, it is an issue that should be addressed and not skipped

Reply:
There is no way to address this issue via openmrs db since the messed data are already there and we cannot resolve right now

Reply:
But the system has been in prod like this for OpenCR, every patient should be pushed to the MPI in opinion otherwise throw an exception so that the record can be fixed first.

Reply:
@danielchirinda why would a patient have no name? We need to find a way to note these patients otherwise we risk having missing patients.

Reply:
I think it's caused by improperly filling out the patient's form.

Reply:
I suggest you submit them to OpenCR with no name rather than skipping them, are you saying Sante rejects them? If yes, we might as well submit them with a place holder like MISSING NAME or NO NAME SPECIFIED

Reply:
What would be the reason to send an unnamed patient to the mpi? What would be the benefit of this data, to me it seems more garbage for the system and we should solve this problem in the source system rather than send it to mpi.

Reply:
Well, it is fine so say don't submit them, because skipping them solves no problem anyways, the only reason I was saying submit them is so that we can look them in the MPI using the place holder and follow up with them for cleaning otherwise if you skip them how will you find them?

Reply:
This is not yet addressed too

Reply:
We will discuss this tomorrow in a weekly meeting.